addons/mail/static/tests/m2x_avatar_user_tests.js

```diff
@@ -58,189 +58,57 @@ QUnit.module('mail', {}, function () {
-            hasView: true,
-            View: ListView,
-            model: 'foo',
-            data: this.data,
-            arch: '<tree editable="top"><field name="user_ids" widget="many2many_avatar_user"/></tree>',
-            res_id: 1,
-        });
-
-        assert.containsN(list, '.o_data_row:eq(0) .o_field_many2manytags.avatar.o_field_widget .o_m2m_avatar', 2,
-            "should have 2 records");
-        assert.containsN(list, '.o_data_row:eq(1) .o_field_many2manytags.avatar.o_field_widget > div > span', 1,
-            "should have 1 record in second row");
-        assert.containsN(list, '.o_data_row:eq(1) .o_field_many2manytags.avatar.o_field_widget > div > span', 1,
-            "should have img and span in second record");
-
-        await dom.click(list.$('.o_data_row:eq(1) .o_field_many2manytags.avatar:first > div > span'));
-        assert.containsOnce(list, '.o_selected_row');
-
-        list.destroy();
-    });
-
-    QUnit.test('many2many_avatar_user widget in list view', async function (assert) {
-        assert.expect(8);
-
-        const { widget: list } = await start({
-            hasView: true,
-            View: ListView,
-            model: 'foo',
-            data: this.data,
-            arch: '<tree><field name="user_ids" widget="many2many_avatar_user"/></tree>',
-            mockRPC(route, args) {
-                if (args.method === 'read') {
-                    assert.step(`read ${args.model} ${args.args[0]}`);
-                }
-                return this._super(...arguments);
-            },
-        });
-
-        mock.intercept(list, 'open_record', () => {
-            assert.step('open record');
-        });
-
-        assert.containsN(list.$(".o_data_cell:first"), '.o_field_many2manytags.avatar.o_field_widget span', 2,
-            "should have 2 records");
-        assert.strictEqual(list.$(".o_data_cell:first .o_field_many2manytags.avatar img.o_m2m_avatar:first").data('src'),
-            "/web/image/res.users/11/avatar_128",
-            "should have right image");
-        assert.strictEqual(list.$(".o_data_cell:eq(0) .o_field_many2manytags.avatar img.o_m2m_avatar:eq(1)").data('src'),
-            "/web/image/res.users/23/avatar_128",
-            "should have right image");
-
-        // sanity check: later on, we'll check that clicking on the avatar doesn't open the record
-        await dom.click(list.$('.o_data_row:first .o_field_many2manytags'));
-
-        await dom.click(list.$('.o_data_cell:nth(0) .o_m2m_avatar:nth(0)'));
-        await dom.click(list.$('.o_data_cell:nth(0) .o_m2m_avatar:nth(1)'));
-
-        assert.verifySteps([
-            'read res.users 11,23',
-            "open record",
-            "read res.users 11",
-            "read res.users 23",
-        ]);
-
-        list.destroy();
-    });
-
     QUnit.test('many2many_avatar_user in kanban view', async function (assert) {
-        assert.expect(11);
+        assert.expect(4);
```

Comment:
Note that this could do even better and split this test in 2: one to assert there is m2m_avatar field widget, and another for popover assertions.

But let's just ignore since this test is out of scope of discuss. It's already a gift to remove complexity.